backend/app/services/source_network.py

```
@@ -982,9 +986,38 @@ async def _lookup_pending(
+            break
+        done, tasks = await asyncio.wait(
+            tasks,
+            timeout=remaining,
+            return_when=asyncio.FIRST_COMPLETED,
+        )
+        if not done:
+            break
+        for task in done:
+            try:
+                looked_up.append(task.result())
+            except Exception as exc:  # pylint: disable=broad-exception-caught
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)